cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -83,16 +83,15 @@ void __device__ inline uint32ToLowercaseHexString(uint32_t num, char* destinatio
 // algorithms are optimized for their respective platforms. You can still
 // compile and run any of them on any platform, but your performance with the
 // non-native version will be less than optimal.
-template <typename Key>
```

Comment:
The precedent in C++ hash function objects is that the type itself is a template and callable operator. 

https://en.cppreference.com/w/cpp/utility/hash

Reply:
@jrhemstad I wasn't familiar with that, thanks for the reference. Do you think we need to retain that class template behavior, or is this comment just informational to explain the previous design? In practice, we don't have any constructor behavior that varies by key type. It seems like a hash functor with type specializations of `operator()` is more in line with the rest of libcudf's design, unless you think that matching `std::hash` is crucial here.

Reply:
I'd prefer to keep the class template. 

> It seems like a hash functor with type specializations of operator() is more in line with the rest of libcudf's design

Class templates are actually more in line with "dispatch to invoke" pattern I've advocated for in the past. See: https://github.com/rapidsai/cudf/pull/8217/files and the Better Code talk I gave about this pattern.